slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/AttestationImporter.java

```diff
@@ -0,0 +1,180 @@
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.jdbi.v3.core.Handle;
+
+public class AttestationImporter {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  final OptionalMinValueTracker minSourceTracker = new OptionalMinValueTracker();
+  final OptionalMinValueTracker minTargetTracker = new OptionalMinValueTracker();
+
+  private final LowWatermarkDao lowWatermarkDao;
+  private final SignedAttestationsDao signedAttestationsDao;
+
+  private final Validator validator;
+  private final Handle handle;
+  private final ObjectMapper mapper;
+
+  public AttestationImporter(
+      final Validator validator,
+      final Handle handle,
+      final ObjectMapper mapper,
+      final LowWatermarkDao lowWatermarkDao,
+      final SignedAttestationsDao signedAttestationsDao) {
+    this.validator = validator;
+    this.handle = handle;
+    this.mapper = mapper;
+    this.lowWatermarkDao = lowWatermarkDao;
+    this.signedAttestationsDao = signedAttestationsDao;
+  }
+
+  public void importFrom(final ArrayNode signedAttestationNode) throws JsonProcessingException {
+
+    for (int i = 0; i < signedAttestationNode.size(); i++) {
+      final SignedAttestation jsonAttestation =
+          mapper.treeToValue(signedAttestationNode.get(i), SignedAttestation.class);
+      final AttestationValidator attestationValidator =
+          new AttestationValidator(
+              handle,
+              validator.getPublicKey(),
+              jsonAttestation.getSigningRoot(),
+              jsonAttestation.getSourceEpoch(),
+              jsonAttestation.getTargetEpoch(),
+              validator.getId(),
+              signedAttestationsDao,
+              lowWatermarkDao);
+
+      // if the attestation is illegal formatted, it cannot be imported
+      final String attesationIdentiferString =
+          String.format("Attestation with index %d for validator %s", i, validator.getPublicKey());
+      if (attestationValidator.sourceGreaterThanTargetEpoch()) {
+        LOG.warn("{} - source is greater than target epoch", attesationIdentiferString);
+      } else {
+
+        if (attestationValidator.isSurroundedByExistingAttestation()) {
+          LOG.warn("{} - is surrounded by existing entries", attesationIdentiferString);
+        }
+
+        if (attestationValidator.surroundsExistingAttestation()) {
+          LOG.warn("{} - surrounds an existing entry", attesationIdentiferString);
+        }
+
+        if (jsonAttestation.getSigningRoot() == null) {
+          if (!nullAttestationAlreadyExistsInTargetEpoch(jsonAttestation.getTargetEpoch())) {
+            persist(jsonAttestation);
+          }
```

Comment:
if we don't meet this condition then it's already in the database? Probably should do "{} - already exists in database, not imported" log in that case

Reply:
done